### PR TITLE
feat: PlayGesture & HotKeys

### PR DESCRIPTION
**Pull request by luwes**
## Summary

Adds **PlayGesture** and **Hotkeys** interaction components.

### PlayGesture

Click/tap the video surface to toggle play/pause — modeled as a declarative gesture component. Built on a `GestureCore` → `PlayGestureCore` abstraction in `@videojs/core` that keeps the logic runtime-agnostic, with DOM binding (`bindGesture`) in `@videojs/core/dom`. Touch events are excluded by default (`type="mouse"`).

### Hotkeys

YouTube-style keyboard shortcuts bound to the media container. `HotkeysCore` maps keys to actions (space/k = play/pause, f = fullscreen, m = mute, arrows = seek/volume, j/l = ±10s seek, `<`/`>` = playback rate). Modifier keys (Ctrl/Meta/Alt) are ignored, and space only fires when the container itself is focused.

### Usage

**HTML** — drop the custom elements inside a skin template:

```html
<media-container>
  <slot name="media"></slot>

  <media-play-gesture></media-play-gesture>
  <media-hotkeys></media-hotkeys>

  <!-- ...controls... -->
</media-container>
```

**React** — render the components inside a `<Container>`:

```tsx
import { Container } from '@videojs/react';
import { PlayGesture } from '@videojs/react/ui/play-gesture';
import { Hotkeys } from '@videojs/react/ui/hotkeys';

function VideoSkin({ children }) {
  return (
    <Container className="my-skin">
      {children}

      <PlayGesture />
      <Hotkeys />

      {/* ...controls... */}
    </Container>
  );
}
```

## References
This PR makes progress towards two sub-issues under the gestures epic (#504)
- #790 — click / tap to play
- #1054 — hotkeys

## Test plan

- [x] `HotkeysCore` unit tests cover all key bindings, modifier exclusion, and edge cases (no media, missing methods)
- [x] `PlayGestureCore` unit tests cover play/pause toggling and ended→play restart
- [ ] Manual: click video surface to toggle playback in HTML and React sandboxes
- [ ] Manual: verify keyboard shortcuts (space, k, f, m, arrows, j, l, <, >) in both sandboxes
- [ ] Manual: confirm touch events do not trigger PlayGesture (mouse-only by default)

### CODE REVIEW

File: packages/core/src/core/ui/gesture/gesture-core.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { defaults } from '@videojs/utils/object';
+
+import type { MediaPlaybackState } from '../../media/state';
+
+export const ALLOWED_TYPES = ['pointerup'] as const;
```
Comment (mihar-22) [inline packages/core/src/core/ui/gesture/gesture-core.ts]: (request): since it's exported from main package entry, recommend `ALLOWED_GESTURE_TYPES` to avoid conflicts or confusion

File: packages/core/src/core/ui/gesture/gesture-core.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+export const ALLOWED_COMMANDS = ['toggle-paused'] as const;
```
Comment (mihar-22) [inline packages/core/src/core/ui/gesture/gesture-core.ts]: (request): same as earlier, recommend `ALLOWED_GESTURE_COMMANDS`

File: packages/core/src/core/ui/gesture/gesture-core.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+export interface GestureProps {
+  type: (typeof ALLOWED_TYPES)[number];
```
Comment (mihar-22) [inline packages/core/src/core/ui/gesture/gesture-core.ts]: Great if we extract into reusable type: `type GestureType = (typeof ALLOWED_GESTURE_TYPES)[number]`

File: packages/core/src/core/ui/gesture/gesture-core.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  command: (typeof ALLOWED_COMMANDS)[number];
```
Comment (mihar-22) [inline packages/core/src/core/ui/gesture/gesture-core.ts]: Great if we extract into reusable type: type `GestureCommand = (typeof ALLOWED_GESTURE_COMMANDS)[number]`

File: packages/html/src/media/container-element.ts [hunk 5]
```diff
@@ -1,9 +1,38 @@
...
 export class MediaContainerElement extends ContainerMixin(MediaElement) {
   static readonly tagName = 'media-container';
+  static shadowRootOptions = { mode: 'open' } as const;
+  static getTemplateHTML(): string {
+    return /* html */ `
+      <style>
+        slot:not([name]) {
+          border-radius: inherit;
+        }
+      </style>
```
Comment (mihar-22) [inline packages/html/src/media/container-element.ts]: This needed?

File: packages/html/src/store/container-mixin.ts
Comment (mihar-22) [inline packages/html/src/store/container-mixin.ts:1]: Just noting this will have docs and consumer impact because they have to use shadow DOM for this mixin to work. Not a problem since children will still live in light DOM, just worth calling out.

File: packages/react/src/ui/gesture/gesture.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        gestureCore.activate(playback);
+      },
+      { signal: controller.signal }
+    );
+
+    return () => controller.abort();
+  }, [playback, gestureCore, type]);
+
+  if (!playback) {
+    if (__DEV__) logMissingFeature('Poster', 'playback');
```
Comment (mihar-22) [inline packages/react/src/ui/gesture/gesture.tsx]: 'Poster' => Gesture?

File: package.json [hunk 7]
```diff
@@ -25,7 +25,7 @@
     "dev:site": "turbo run dev --filter=site",
     "dev": "turbo run dev --parallel",
     "dev:packages": "turbo run dev --parallel --filter='./packages/*'",
-    "dev:sandbox": "turbo run dev --filter=@videojs/sandbox",
```
Comment (mihar-22) [inline package.json]: Interesting, what's this change do?

File: package.json [hunk 8]
```diff
@@ -25,7 +25,7 @@
...
+    "dev:sandbox": "turbo watch dev --filter=@videojs/sandbox",
```
Comment (mihar-22) [inline package.json:29]: Interesting, what's this change do?

File: packages/html/src/media/container-element.ts (hunk 5 above)
Comment (luwes) [reply packages/html/src/media/container-element.ts]: it was needed if we used shadow dom in media container because we used border-radius inherit on the media (I think) so this was needed. I have reverted this logic to not use shadow dom...

File: packages/html/src/store/container-mixin.ts
Comment (luwes) [reply packages/html/src/store/container-mixin.ts:1]: reverted and updated

File: package.json (hunk 8 above)
Comment (luwes) [reply package.json:29]: this will watch the scripts that are not never ending, in this case the build scripts of the dependencies

File: packages/core/src/core/ui/gesture/gesture-core.ts [hunk 9]
```diff
@@ -0,0 +1,59 @@
+import { defaults } from '@videojs/utils/object';
+
+import type { MediaPlaybackState } from '../../media/state';
+
+export type GestureType = (typeof ALLOWED_GESTURE_TYPES)[number];
+export type GestureCommand = (typeof ALLOWED_GESTURE_COMMANDS)[number];
+
+export interface GestureProps {
+  type: GestureType;
+  command: GestureCommand;
```
Comment (mihar-22) [inline packages/core/src/core/ui/gesture/gesture-core.ts]: (request): missing jsdoc comments, used by API reference tooling for docs.

File: packages/core/src/core/ui/gesture/gesture-core.ts [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+export const ALLOWED_GESTURE_TYPES = ['pointerup'] as const;
+export const ALLOWED_GESTURE_COMMANDS = ['toggle-paused'] as const;
+
+export const PointerTypes = {
+  MOUSE: 'mouse',
+  PEN: 'pen',
+  TOUCH: 'touch',
+} as const;
```
Comment (mihar-22) [inline packages/core/src/core/ui/gesture/gesture-core.ts]: (request): this is handy but more a DOM specific type. Can we move this to `core/ui/types`, or as per suggestion below have `core/dom` layer for gesture and move to `core/dom/ui/event`

File: packages/core/src/core/ui/gesture/gesture-core.ts [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+    this.#media = media;
+  }
+
+  handleGesture({ pointerType }: { pointerType: string }): void {
+    if (!this.#media) return;
+
+    if (!ALLOWED_GESTURE_TYPES.includes(this.#props.type)) return;
+
+    // TODO: Should `pointerType` be a prop that can be configured?
+    if (pointerType && pointerType === PointerTypes.MOUSE) {
```
Comment (mihar-22) [inline packages/core/src/core/ui/gesture/gesture-core.ts]: Interesting. I like it and if it makes sense for future gestures then good idea! 

(suggestion): if it doesn't make sense as a prop, you could consider moving "pointer type" handling to the DOM layer or framework-layer since it's just a DOM event related guard and not core state management piece.

File: packages/core/src/core/ui/gesture/gesture-core.ts [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+
+  setProps(props: GestureProps): void {
+    this.#props = defaults(props, GestureCore.defaultProps);
+  }
+
+  setMedia(media: MediaPlaybackState): void {
+    this.#media = media;
+  }
+
+  handleGesture({ pointerType }: { pointerType: string }): void {
```
Comment (mihar-22) [inline packages/core/src/core/ui/gesture/gesture-core.ts]: (request): stricter type `pointerType: PointerType`. 

Requires creating the type:

```ts
type PointerType = PointerTypes[typeof keyof PointerTypes]
```

---

Not required if guarding is moved to `core/dom`

File: packages/html/src/ui/gesture/gesture-element.ts [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+          (event: PointerEvent) => {
+            const target = event.target as Element;
+            if (target !== container && !target.localName.endsWith('video')) return;
+
+            this.#core.handleGesture(event);
+          },
+          { signal }
+        );
+      }
+    }
```
Comment (mihar-22) [inline packages/html/src/ui/gesture/gesture-element.ts]: (suggestion): could move this to `core/dom` to share with html and react. The pointer types could live there too which would make more sense. Claude should be able to do it by referencing other examples.

File: packages/react/src/ui/gesture/gesture.tsx [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+
+export function Gesture({ type, command }: GestureProps) {
+  const [gestureCore] = useState(() => new GestureCore());
+  gestureCore.setProps({ type, command });
+
+  const playback = usePlayer(selectPlayback);
+  if (playback) gestureCore.setMedia(playback);
+  else {
+    if (__DEV__) logMissingFeature('Gesture', 'playback');
+  }
```
Comment (mihar-22) [inline packages/react/src/ui/gesture/gesture.tsx]: (nit): could be `else if (__DEV__)`
